flask_marshmallow/sqla.py

```diff
@@ -41,31 +42,30 @@ class ModelSchema(msqla.ModelSchema, Schema):
     """
     OPTIONS_CLASS = SchemaOpts
 
-def hyperlink_keygetter(obj):
-    link_attribute = current_app.config['MARSHMALLOW_LINK_ATTRIBUTE']
-    try:
-        return getattr(obj, link_attribute)
-    except AttributeError:
-        # Reraise with better error message
-        raise AttributeError(
-            'Objects that get serialized by HyperlinkModelSchema must '
-            'have a "{0}" attribute.'.format(link_attribute)
-        )
-
-class HyperlinkSchemaOpts(SchemaOpts):
-    def __init__(self, meta):
-        if not hasattr(meta, 'keygetter'):
-            meta.keygetter = hyperlink_keygetter
-        super(HyperlinkSchemaOpts, self).__init__(meta)
+class HyperlinkRelated(msqla.fields.Related):
 
+    def __init__(self, endpoint, url_key='id', external=False, **kwargs):
+        super(HyperlinkRelated, self).__init__(**kwargs)
+        self.endpoint = endpoint
+        self.url_key = url_key
+        self.external = external
 
-class HyperlinkModelSchema(msqla.ModelSchema):
-    """A `ModelSchema <marshmallow_sqlalchemy.ModelSchema>` that serializes relationships
-    to hyperlinks. Related models MUST have a ``url`` attribute or property (unless the
-    ``MARSHMALLOW_LINK_ATTRIBUTE`` app config option is set).
```

Comment:
If we're getting rid of this, then the `MARSHMALLOW_LINK_ATTRIBUTE` on ~L121 can be removed.